hail/python/hailtop/utils/gcs_requester_pays.py

```diff
@@ -0,0 +1,25 @@
+from typing import Dict, Optional, Type, TypeVar
+
+from hailtop.aiocloud.aiogoogle import GCSRequesterPaysConfiguration
+
+
+FS = TypeVar("FS")
+
+
+def get_fs_by_requester_pays_config(
+    cls: Type[FS],
+    requester_pays_fses: Dict[Optional[str], FS],
+    requester_pays_config: Optional[GCSRequesterPaysConfiguration] = None
+) -> FS:
+    hashable_config = (
+        requester_pays_config
+        if requester_pays_config is None or isinstance(requester_pays_config, str)
+        else f"{requester_pays_config[0]}:{','.join(requester_pays_config[1])}"
```

Comment:
I think this will produce different keys for `("project", ["bucket1", "bucket2"])` vs `("project", ["bucket2", "bucket1"])`. If doing this interpolation is stemming from lists not being hashable, you can construct a `tuple(str, frozenset)` which should be hashable.

Reply:
ooo good catch, updated to use a `frozenset`